homeassistant/components/unifiprotect/event.py

```diff
@@ -167,6 +175,125 @@ def _async_update_device_from_protect(self, device: ProtectDeviceType) -> None:
+        event = self.entity_description.get_event_obj(self.device)
+        if not event or event.id != event_id:
+            return
+
+        # Get current vehicle thumbnails
+        current_thumbnails: list[EventDetectedThumbnail] = []
+        if event.metadata and event.metadata.detected_thumbnails:
+            current_thumbnails = [
+                t for t in event.metadata.detected_thumbnails if t.type == "vehicle"
+            ]
+
+        if not current_thumbnails:
+            return
+
+        # Start with just the event ID
+        event_data: dict[str, Any] = {
+            ATTR_EVENT_ID: event.id,
+        }
+
+        # Select best thumbnail
+        # Prefer thumbnails with LPR data, sorted by confidence
+        # LPR can be in: 1) group.matched_name (UFP 6.0+) or 2) name field
+        def _thumbnail_sort_key(t: EventDetectedThumbnail) -> tuple[bool, float, float]:
+            """Sort key: (has_lpr, confidence, clock_best_wall)."""
+            has_lpr = bool(
+                (t.group and t.group.matched_name) or (t.name and len(t.name) > 0)
+            )
+            confidence = t.confidence if t.confidence else 0.0
+            clock = t.clock_best_wall.timestamp() if t.clock_best_wall else 0.0
+            return (has_lpr, confidence, clock)
+
+        thumbnail = max(current_thumbnails, key=_thumbnail_sort_key)
+
+        # Add confidence if available
+        if thumbnail.confidence is not None:
+            event_data["confidence"] = thumbnail.confidence
+
+        # Add best detection frame timestamp
+        if thumbnail.clock_best_wall is not None:
+            event_data["clock_best_wall"] = thumbnail.clock_best_wall.isoformat()
+
+        # License plate from group.matched_name (UFP 6.0+) or name field (older)
+        if thumbnail.group and thumbnail.group.matched_name:
+            event_data["license_plate"] = thumbnail.group.matched_name
+        elif thumbnail.name:
+            event_data["license_plate"] = thumbnail.name
+
+        # Add all thumbnail attributes as dict
+        if thumbnail.attributes:
+            event_data["attributes"] = thumbnail.attributes.unifi_dict()
+
+        self._trigger_event(EVENT_TYPE_VEHICLE_DETECTED, event_data)
+        self.async_write_ha_state()
+
+        # Clear pending event tracking and timer
+        self._pending_event_id = None
+        self._cancel_thumbnail_timer()
+
+    @callback
+    def _async_update_device_from_protect(self, device: ProtectDeviceType) -> None:
+        description = self.entity_description
+
+        super()._async_update_device_from_protect(device)
+        if event := description.get_event_obj(device):
+            self._event = event
+            self._event_end = event.end if event else None
+
+        # Check if detected_thumbnails just arrived for this event
+        if event and event.type is EventType.SMART_DETECT:
+            # Get current vehicle thumbnails
+            current_thumbnails: list[EventDetectedThumbnail] = []
```

Comment:
Fetching current thumbnails looks like it's duplicated in the fire vehicle event above and could be a helper